earthpy/clip.py

```diff
@@ -146,14 +241,14 @@ def clip_shp(shp, clip_obj):
     if not any(shp.intersects(clip_obj.unary_union)):
         raise ValueError("Shape and crop extent do not overlap.")
 
-    # Multipolys / point / line don't clip properly
-    if "Multi" in str(clip_obj.geom_type) or "Multi" in str(shp.geom_type):
-        raise ValueError(
-            "Clip doesn't currently support multipart geometries. Consider "
-            "using .explode to create unique features in your GeoDataFrame"
-        )
-
-    if shp["geometry"].iloc[0].type == "Point":
+    # if there are multi points
+    if any(shp.geometry.type == "MultiPoint"):
+        return _clip_multi_point(shp, clip_obj)
+    elif shp["geometry"].iloc[0].type == "Point":
```

Comment:
I'm noticing some places where we use `shp["geometry"].iloc[0].type` instead of `shp.geometry.type`. The former seems somewhat brittle because we only look at the first row. Is there a good reason for us to do this, or should we change this to 

```python
elif any(shp.geometry.type == "Point"):
```

Reply:
hmmm i suspect this is just a product of us working and reworking this for so long. it should be `any()` as that is much more robust!!

Reply:
actually this is an interesting potential glitch. if for some reason someone has a mix of shapes this could fail. this is very atypical as a shapefile can't support that BUT geopandas does support this. maybe this is a see what happens situation because it's an edge case but i could see it happening if someone does some funky processing first. i'll switch to any for the time being and then wait for any edge cases to come in as bugs IF they come in. in theory it shouldn't be an issue if people are using conventional shapefiles. BUT not sure about geojson support.